doc/Language/variables.pod6

```diff
@@ -498,6 +399,212 @@ augment slang Regex {  # derive from $~Regex and then modify $~Regex
+    my ($f, $g, $h, $i) = (1, 2, 3); # list assignment to four Scalars
+    say $f; say $g; say $h; say $i;  # OUTPUT:  «1␤2␤3␤(Any)␤»
+
+    # @-sigiled variables take all remaining values
+    my ($j, @k, $l) = (1, 2, 3);
+    say $j; say @k; say $l;      # OUTPUT:  «1␤[2 3]␤(Any)␤»
+
+    # so do %-sigiled variables
+    my ($m, %n, $o) = (1, 2, 3);
+    say $m; say %n; say $o;      # OUTPUT:  «1␤{2 => 3}␤(Any)␤»
+
+    # if a %-sigiled variable can't construct a Hash, you get an error
+    try my ($q, %r) = (1, 2);
+    say $!;
+    # OUTPUT: «Odd number of elements found where hash initializer expected:␤
+    #          Only saw: 2»
+
+
+Both item assignment and list assignment use the equal sign (C<=>) as their
+operator and are both right associative, but differ in L<operator
+precedence|/language/operators#Operator_precedence>: item assignment has a higher
+precedence level (level: Item assignment) than list assignment (level: List
+prefix).
+
+This distinction can matter when your right hand side contains a comma-separated
+list of elements because item assignment has tighter precedence than the comma
+operator C<,> while list assignment has a looser precedence.  Note, however, that
+this won't matter if you use parentheses on the left hand side, which remove the
+need to consider precedence.
+
+    my $a = 1, 2, 3; # item assignment; higher precedence than ,
+    say $a;          # OUTPUT: «1␤»
+
+    my @b = 1, 2, 3; # list assignment; lower precedence than ,
+    say @b;          # OUTPUT: «[1 2 3]␤»
+
+    my ($c) = 1, 2, 3; # list assignment; lower precedence than , but using ( )
+    say $c;            # OUTPUT:  «1␤»
+
+    # rare but valid syntax
+    (my $d) = 1, 2, 3; # list assignment; lower precedence than ,
+    say $d;            # OUTPUT: «(1 2 3)␤»
+
+    # an initial ( also lets you add a , which stops @ and % from taking all values
+    (my (@e, $f, $g),) = 1, 2, 3;
+    say @e; say $f; say $g;       # OUTPUT: «[1]␤2␤3»
+
+The precedence of the list and item assignment also matters for chained
+assignments. For instance, in the example below there is one chained assignment
+statement comprising two assignment operators. The assignment to C<@array> is a
+list assignment having a lower precedence than the item assignment to the Scalar
+variable C<$num>. The assignment expression involving the item assignment to the
+Scalar variable C<$num> is thus evaluated first. It returns the assigned value
+C<42>, which in turn forms part of the List C<(42, "str")> constructed by the
+comma operator that also has a higher precedence than the list
+assignment. Finally, the List C<(42, "str")> is list-assigned to C<@array>:
+
+    #`[this statement:] my @array = my $num = 42, "str";
+    # is parsed as      my @array = ( (my $num = 42), "str" )
+    say @array.raku;    # OUTPUT: «[42, "str"]␤» (an Array)
+    say $num.raku;      # OUTPUT: «42␤» (an Int)
+
+Here's a variant:
+
+    #`[this:]      my @foo = (my $bar) = 42, "str";
```

Comment:
OK, literary programming with embedded comments used here... This might be confusing for newcomers. Just put it in different lines.